scalding-macros/src/test/scala/com/twitter/scalding/macros/MacroOrderingProperties.scala

```diff
@@ -0,0 +1,540 @@
+  }
+
+  def checkAreSame[T](a: T, b: T)(implicit obuf: OrderedSerialization[T]) {
+    val rta = rt(a) // before we do anything ensure these don't throw
+    val rtb = rt(b) // before we do anything ensure these don't throw
+    assert(oBufCompare(rta, a) === 0, s"A should be equal to itself after an RT -- ${rt(a)}")
+    assert(oBufCompare(rtb, b) === 0, "B should be equal to itself after an RT-- ${rt(b)}")
+    assert(oBufCompare(a, b) === 0, "In memory comparasons make sense")
+    assert(oBufCompare(b, a) === 0, "In memory comparasons make sense")
+    assert(rawCompare(a, b) === 0, "When adding the raw compares in inverse order they should sum to 0")
+    assert(rawCompare(b, a) === 0, "When adding the raw compares in inverse order they should sum to 0")
+    assert(oBufCompare(rta, rtb) === 0, "Comparing a and b with ordered bufferables compare after a serialization RT")
+  }
+
+  def check[T: Arbitrary](implicit obuf: OrderedSerialization[T]) = {
+    Checkers.check(LawTester(OrderedSerialization.allLaws))
+    forAll(minSuccessful(500)) { (a: T, b: T) => checkWithInputs(a, b) }
+  }
+
+  test("Test out Unit") {
+    primitiveOrderedBufferSupplier[Unit]
+    check[Unit]
+    checkMany[Unit]
+  }
+  test("Test out Boolean") {
+    primitiveOrderedBufferSupplier[Boolean]
+    check[Boolean]
+  }
+  test("Test out jl.Boolean") {
+    implicit val a = arbMap { b: Boolean => java.lang.Boolean.valueOf(b) }
+    check[java.lang.Boolean]
+  }
+  test("Test out Byte") { check[Byte] }
+  test("Test out jl.Byte") {
+    implicit val a = arbMap { b: Byte => java.lang.Byte.valueOf(b) }
+    check[java.lang.Byte]
+  }
+  test("Test out Short") { check[Short] }
+  test("Test out jl.Short") {
+    implicit val a = arbMap { b: Short => java.lang.Short.valueOf(b) }
+    check[java.lang.Short]
+  }
+  test("Test out Char") { check[Char] }
+  test("Test out jl.Char") {
+    implicit val a = arbMap { b: Char => java.lang.Character.valueOf(b) }
+    check[java.lang.Character]
+  }
+  test("Test out Int") {
+    primitiveOrderedBufferSupplier[Int]
+    check[Int]
+    checkMany[Int]
+  }
+  test("Test out jl.Integer") {
+    implicit val a = arbMap { b: Int => java.lang.Integer.valueOf(b) }
+    check[java.lang.Integer]
+  }
+  test("Test out Float") { check[Float] }
+  test("Test out jl.Float") {
+    implicit val a = arbMap { b: Float => java.lang.Float.valueOf(b) }
+    check[java.lang.Float]
+  }
+  test("Test out Long") { check[Long] }
+  test("Test out jl.Long") {
+    implicit val a = arbMap { b: Long => java.lang.Long.valueOf(b) }
+    check[java.lang.Long]
+  }
+  test("Test out Double") { check[Double] }
+  test("Test out jl.Double") {
+    implicit val a = arbMap { b: Double => java.lang.Double.valueOf(b) }
+    check[java.lang.Double]
+  }
+
+  test("Test out String") {
+    primitiveOrderedBufferSupplier[String]
+
+    check[String]
```

Comment:
so that approach, as best I can tell, would generate strings with invalid characters in it. Not all 16 bit numbers are valid jvm characters -- some of them are only valid when immediately followed by certain other characters.

http://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#unicode

If scalacheck gets around this by filtering those out, then we might not be testing a particular type of character. I'm probably overthinking this though.